Added documentation to the es_fine-tuning_conciseness.py

kxk302 opened the pull request:
Added source code formatting (isort and black). 
Added linting tools (flake8 and pylint) to catch Python lint errors.

Applied source code formatting and linting to all Python files. 

Also, layer, must rename the Python files that have '-' on their names (replaced '-' with '_' to make sure the file name follows the standard, where '-' is not allowed). Must also delete scripts with "iid" in the name as they are nearly identical to original scripts (functionality can be added to the original scripts via a new CLI argument) 

Added markdown linting to catch markdown lint errors
Updated the documentation with instructions on how to contribute to the repo (fork, create feature branch, run linting/formatting, create PR).

@VsonicV, @dibbla, @sunrainyg for review please

kxk302 committed:
Documented the module. Ran isort, black, flake8, and pylint and fixed the ensuing issues.

kxk302 committed:
Cannot have dash in file name. Replaced with underscore

kxk302 committed:
Added README for code quality plus other files needed

kxk302 committed:
markdown formatting instructions

kxk302 committed:
Documented half of the module. The rest to follow

kxk302 committed:
Initial documentation for accl script. Ran isort and black. Must run linting next

kxk302 committed:
Documented the accelerated version of the script and related scripts. Ran black, isort, pylint, and flake8

kxk302 committed:
Ran isort, black, flake8, and pylint and fixed issues

kxk302 committed:
Added a flag to es_fine_tuning_conciseness to accomodate seed shifting. No need to the _iid version of the script

kxk302 committed:
Added a flag to es_fine_tuning_countdown to accomodate seed shifting. No need to the _iid version of the script

kxk302 committed:
Added yaml file for GitHub acrtions (Linting, formatting). Minor cleanup

VsonicV reviewed (changes requested):
Two requests before further review:
1. Please do not change file names in this PR, as it 1) makes the file changes summary difficult to catch (it displays the change as an addition of an entirely new file), though we can still see the history by checking each commit. 2) it makes the command examples in README invalid. Let's change the file names later together with README.
2. Please do the combination of iid version and non-iid version in another PR. This PR should only be adding documentations and trivial reformattings.

kxk302 committed:
Added these files back per Xin's request

kxk302 committed:
Renamed files so the diff can be done

kxk302 committed:
Resolved merge conflict

kxk302 committed:
Temporarily removed seed_shift flag. Must revert this change in a separate PR, and delete iid scripts afterwards

kxk302 committed:
Updated contribution doc. Fixed minor formatting issues

kxk302 commented:
> Two requests before further review:
> 
> 1. Please do not change file names in this PR, as it 1) makes the file changes summary difficult to catch (it displays the change as an addition of an entirely new file), though we can still see the history by checking each commit. 2) it makes the command examples in README invalid. Let's change the file names later together with README.
> 2. Please do the combination of iid version and non-iid version in another PR. This PR should only be adding documentations and trivial reformattings.

@VsonicV, 

1) I changed back the file name so the diffs can be reviewed easily. After this PR is merged, we have to rename the files again, as Python standard does not allow '-' in file names (Only '_' is allowed). E.g., you cannot do "import es-fine-tuning". It will error out. But you can do "import es_fine_tuning". Good catch on the documentation being invalid. After we re-name the file, we need to update the docs as well.

2) I added the iid version of the scripts back to the repo, undid the change in the regular version of the script that handles seed shifting. Again, after this PR is merged, I suggest we delete the iid version of the script and re-add the seed shifting functionality to the regular version of the scripts. We should avoid having 2 scripts that share 99% of the same functionality 

3) I updated the contribution README file and specified the steps for contributing to the repo. It is important that all the team members follow the same steps, to ensure any new code is lint error free, is formatted correctly, and is reviewed before being merged.

kxk302 committed:
Updated yaml file so all tests run and if there is a fialure on any of them, the GitHub Actions fail
